utils/src/test/kotlin/storage/LocalFileStorageTest.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.utils.storage
+
+import com.here.ort.utils.safeMkdirs
+
+import io.kotlintest.shouldBe
+import io.kotlintest.shouldNotThrowAny
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.WordSpec
+
+import java.io.BufferedReader
+import java.io.File
+
+class LocalFileStorageTest : WordSpec() {
+
+    private fun storage(block: (LocalFileStorage, File) -> Unit) {
+        val directory = createTempDir(javaClass.simpleName)
+        val storage = LocalFileStorage(directory)
+        block(storage, directory)
+    }
+
+    init {
+        "Creating the storage" should {
+            "succeed if the directory exists" {
+                shouldNotThrowAny {
+                    LocalFileStorage(createTempDir(javaClass.simpleName))
+                }
+            }
+
+            "succeed if the directory does not exist and shall be created" {
+                val directory = createTempDir(javaClass.simpleName)
+                val storageDirectory = directory.resolve("create/storage")
+
+                LocalFileStorage(storageDirectory, createDirectory = true)
+
+                storageDirectory.isDirectory shouldBe true
+            }
+
+            "fail if the directory is a file" {
+                val storageDirectory = createTempFile()
+
+                shouldThrow<IllegalArgumentException> {
+                    LocalFileStorage(storageDirectory)
+                }
+            }
+
+            "fail if the directory does not exist and shall not be created" {
```

Comment:
how about "fail and not create the directory if the directory does not exist"

Reply:
I think my name better expresses the use of the `createDirectory` parameter.

Reply:
let's just keep the name then.